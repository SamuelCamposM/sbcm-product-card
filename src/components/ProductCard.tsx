import React from 'react';
import styles from '../styles/styles.module.css';
import useProduct from '../hooks/useProduct';
import { createContext } from 'react';

import {
  InitialValues,
  Product,
  ProductCardHandlers,
  ProductContextProps,
  onChangeArgs,
} from '../interfaces/ProductsInterfaces';

export const ProductContext = createContext({} as ProductContextProps);
const { Provider } = ProductContext;
export interface Props {
  product: Product;
  // children?: ReactElement | ReactElement[];
  children: (args: ProductCardHandlers) => JSX.Element;
  className?: string;
  style?: React.CSSProperties;
  onChange?: (args: onChangeArgs) => void;
  value?: number;
  initialValues?: InitialValues;
}

export const ProductCard = ({
  children,
  product,
  className,
  style,
  onChange,
  value,
  initialValues,
}: Props) => {
  // const [counter, setcounter] = useState(0);
  // const increaseBy = (value: number) => {
  // setcounter((prev) => Math.max(prev + value, 0));
  // };
  const {
    counter,
    increaseBy,
    maxCount,
    isMaxCountReached,
    reset,
  } = useProduct({
    onChange,
    product,
    value,
    initialValues,
  });
  return (
    <Provider
      value={{
        counter,
        increaseBy,
        product,
        maxCount,
      }}
    >
      <div style={style} className={`${styles.productCard} ${className}`}>
        {children({
          count: counter,
          isMaxCountReached: isMaxCountReached,
          maxCount: initialValues?.maxCount,
          product,
          increaseBy,
          reset,
        })}
      </div>
    </Provider>
  );
};
