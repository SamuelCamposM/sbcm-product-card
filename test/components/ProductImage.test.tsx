import React from 'react';
import rendered from 'react-test-renderer';
import { ProductCard, ProductImage } from '../../src/components';
import { product2 } from '../data/products';

describe('ProductImage', () => {
  test('debe de mostrar el componente correctamente con la imagen personalizada', () => {
    const wrapper = rendered.create(
      <ProductImage img="https://hola.jpg" className="custom-class" />
    );
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
  test('Debe de mostrar el componente con la imagen del produccto', () => {
    const wrapper = rendered.create(
      <ProductCard product={product2}>
        {() => {
          return <ProductImage />;
        }}
      </ProductCard>
    );
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
