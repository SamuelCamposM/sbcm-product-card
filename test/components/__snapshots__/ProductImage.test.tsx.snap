// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductImage Debe de mostrar el componente con la imagen del produccto 1`] = `
<div
  className="productCard undefined"
>
  <img
    alt="Product"
    className="productImg undefined"
    src="./coffee-mug.png"
  />
</div>
`;

exports[`ProductImage debe de mostrar el componente correctamente con la imagen personalizada 1`] = `
<img
  alt="Product"
  className="productImg custom-class"
  src="https://hola.jpg"
/>
`;
